ShopApp/Factory/Input.cs

```diff
@@ -0,0 +1,281 @@
+﻿using GroupProject.DTO.Enums;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GroupProject.factory
+{
+    public class Input
+    {
+        public static string GetName()
+        {
+            Console.WriteLine("Enter product name: ");
+            string res = Console.ReadLine();
+            if (res.Length > 2) { return res; }
+            return GetName();
+        }
+
+        public static ProdType GetProductType()
+        {
+            Console.WriteLine("Product Type ( 0 - Food, 1 - Meat, 2 - Vehicle): ");
+            string input = Console.ReadLine();
+
+            try
+            {
+                uint res = Convert.ToUInt32(input);
+                switch (res)
+                {
+                    case 0:
+                        return (ProdType)0;
+                    case 1:
+                        return (ProdType)1;
+                    case 2:
+                        return (ProdType)2;
+
+                    default:
+                        Console.WriteLine("Wrong number.");
+                        return GetProductType();
+                }
+            }
+            catch(FormatException)
+            {
+                Console.WriteLine("Incorrect formatting.");
+                return GetProductType();
+            }
+            catch (OverflowException ex)
+            {
+                Console.WriteLine(ex.Message);
+                return GetProductType();
+            }
+        }
+
+        public static decimal GetPrice()
+        {
+            try
+            {
+                Console.WriteLine("Enter product price: ");
+                decimal res = Convert.ToDecimal(Console.ReadLine());
+
+                if (res > 0) { return res; }
+                Console.WriteLine("Price cannot be negative.");
+                return GetPrice();
+            }
+            catch (FormatException)
+            {
+                Console.WriteLine("Incorrect formatting.");
+                return GetPrice();
+            }
+            catch (OverflowException ex)
+            {
+                Console.WriteLine(ex.Message);
+                return GetPrice();
+            }
+        }
+
+        public static double GetWeight()
+        {
+            try
+            {
+                Console.WriteLine("Enter product weight: ");
+                double res = Convert.ToDouble(Console.ReadLine());
+
+                if (res > 0) { return res; }
+                return GetWeight();
+            }
+            catch (FormatException)
+            {
+                Console.WriteLine("Incorrect formatting.");
+                return GetWeight();
+            }
+            catch (OverflowException ex)
+            {
+                Console.WriteLine(ex.Message);
+                return GetWeight();
+            }
+        }
```

Comment:
For the future, all these methods should be inside fabric with product type you want to use one.
Get meatType with MeatFabricConsole etc.